### PR TITLE
fix: fix takePhoto not working inside an Modal

### PR DESCRIPTION
**Pull request by eduardoborges**
## What

When the camera view is inside of Modal they dont take photo, throws `[system/view-not-found]`.

## Changes

Changed way how Camera view is resolved.
We utilize the global React Native View registry to resolve the view instance.

## Tested on

Android 10 emulator
Android 10 device

### CODE REVIEW

**Comment by mrousavy** [comment]
Hey! Thanks for your PR!

Doesn't this throw a "main thread" error? Did you test this without issues?

**Comment by mrousavy** [comment]
We now probably need to call `findCameraView` using withContext to ensure it's being called from Main/UI Thread.

**Review by erwanferry — APPROVED** [review]

**Comment by eduardoborges** [comment]
> We now probably need to call `findCameraView` using withContext to ensure it's being called from Main/UI Thread.

OK, but... Who to do this? LOL

**Comment by PABourdais** [comment]
Wow nice that did fixed the `takePhoto`, I just tested it  👍

**Comment by eduardoborges** [comment]
@mrousavy Can you help-me in your change request?

**Comment by sunnhyn** [comment]
I met same problem and fixed with this PR. thanks!

Here is patch file for [patch-package](https://github.com/ds300/patch-package)
[react-native-vision-camera+2.13.3.patch](https://gist.github.com/StationSoen/6b3f5f9dd2002384abb7d22e402f51fe)

File: android/src/main/java/com/mrousavy/camera/CameraViewModule.kt [hunk 1]
```diff
@@ -75,7 +81,14 @@ class CameraViewModule(reactContext: ReactApplicationContext) : ReactContextBase
     return TAG
   }
 
-  private fun findCameraView(id: Int): CameraView = reactApplicationContext.currentActivity?.findViewById(id) ?: throw ViewNotFoundError(id)
+  private fun findCameraView(viewId: Int): CameraView {
+    Log.d(TAG, "Finding view $viewId...")
+    val mContext = WeakReference(reactApplicationContext)
+    val ctx = mContext?.get()
```
Comment (mrousavy) [inline android/src/main/java/com/mrousavy/camera/CameraViewModule.kt]: Hey - I think this is unnecessary. We should just be able to pass in `reactApplicationContext` to the function below without having to create a WeakReference and access it immediately afterwards.

**Review by mrousavy — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/mrousavy/camera/CameraViewModule.kt [hunk 2]
```diff
@@ -83,8 +82,7 @@ class CameraViewModule(reactContext: ReactApplicationContext) : ReactContextBase
 
   private fun findCameraView(viewId: Int): CameraView {
     Log.d(TAG, "Finding view $viewId...")
-    val mContext = WeakReference(reactApplicationContext)
-    val ctx = mContext?.get()
+    val ctx = reactApplicationContext
     val view = if (ctx != null) UIManagerHelper.getUIManager(ctx, viewId)?.resolveView(viewId) as CameraView? else null
```
Comment (mrousavy) [inline android/src/main/java/com/mrousavy/camera/CameraViewModule.kt]: Can you remove `ctx` and just use `reactApplicationContext` directly? I see no point in declaring an extra variable.

**Comment by eduardoborges** [comment]
@mrousavy Can u see this PR? <3

**Comment by mrousavy** [comment]
Oh I'm sorry I missed this! It somewhere got swallowed by thousands of other emails 😅 

LGTM, thank you so much for your help here!

**Comment by master-eugene** [comment]
@mrousavy publish this fix please

**Comment by ajithes1** [comment]
is this published? I'm still facing this issue. @mrousavy

**Pull request merged**